src/backend/compiler/compiler.cc

```diff
@@ -990,38 +991,69 @@ extern "C" void seashell_preprocessor_free(struct seashell_preprocessor *preproc
 }
 
 /**
- * seashell_preprocessor_set_main_file(struct seashell_preprocessor* preprocessor, const char* file)
- * Sets the main file to begin running the preprocessor on
+ * seashell_preprocessor_set_question_dir(struct seashell_preprocessor* preprocessor, const char *dir)
```

Comment:
Sure, that makes sense.  I'm only asking as getting the containing directory for a header file is relatively straightforward with the clang API.